src/main/java/com/palantir/gradle/gitversion/VersionDetailsImpl.java

```diff
@@ -47,63 +46,25 @@ public String getVersion() {
         if (description() == null) {
             return "unspecified";
         }
-
         return description() + (isClean() ? "" : ".dirty");
     }
 
     private boolean isClean() {
-        try {
-            return git.status().call().isClean();
-        } catch (GitAPIException e) {
-            throw new RuntimeException(e);
-        }
+        return nativeGitInvoker.isClean();
     }
 
     private String description() {
-        if (maybeCachedDescription != null) {
-            return maybeCachedDescription;
-        }
-
         String rawDescription = expensiveComputeRawDescription();
-        maybeCachedDescription =
+        String processedDescription =
                 rawDescription == null ? null : rawDescription.replaceFirst("^" + args.getPrefix(), "");
-        return maybeCachedDescription;
+        return processedDescription;
     }
 
     private String expensiveComputeRawDescription() {
-        if (isRepoEmpty()) {
-            log.debug("Repository is empty");
-            return null;
-        }
-
-        String nativeGitDescribe = new NativeGitDescribe(git.getRepository().getDirectory()).describe(args.getPrefix());
-        String jgitDescribe = new JGitDescribe(git).describe(args.getPrefix());
-
-        // If native failed, return JGit one
-        if (nativeGitDescribe == null) {
```

Comment:
Hmmm, this actually ends up using the jgit version if the native one fails - I misremembered and thought that it was always using the native one. I think we need to think about this carefully - actually removing jgit will cause this to fail in environments where the `git` binary is not installed.

Reply:
Discussed internal and it makes sense to get rid of jgit